src/main/java/org/jenkinsci/plugins/github_branch_source/JwtHelper.java

```diff
@@ -10,12 +10,17 @@
 import java.security.spec.PKCS8EncodedKeySpec;
 import java.util.Base64;
 import java.util.Date;
-import java.util.Objects;
 
 import static java.util.Objects.requireNonNull;
+import java.util.concurrent.TimeUnit;
 
 class JwtHelper {
 
+    /**
+     * @see <a href="https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#authenticating-as-a-github-app">maximum JWT validity</a>
+     */
+    static final long VALIDITY_MS = TimeUnit.MINUTES.toMillis(10);
```

Comment:
This is too tight, we had a low number of requests failing due to clock skew, see stacktrace and PR here: https://github.com/jenkinsci/github-branch-source-plugin/pull/269

Reply:
By “too tight” do you mean too _large_, i.e. too close to 10m? I can restore it to 8m if there is some reason to. Was just following the advice offered by GH.

Reply:
> see stacktrace

Where?

Reply:
ah sorry https://github.com/jenkinsci/github-branch-source-plugin/pull/288